### PR TITLE
Fix #10914 - Crash on printing struct format

### LINKED ISSUE #10914: Crash on printing struct format string without specifying the name
### Work environment

specifying

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Ubuntu 16.04 x86_64, and MacOS 10.13.4
| File format of the file you reverse (mandatory)      | ELF
| Architecture/bits of the file (mandatory)            | x86/64
| r2 -v full output, **not truncated** (mandatory)     | radare2 2.8.0-git 18970 @ linux-x86-64 git.2.7.0-244-gc66112c commit: c66112c13bd59ecb1dcaaadea8106b6b44164f0c build: 2018-08-03__19:03:15

### Expected behavior

Not crashing when typing `ts*` without the struct name.

### Actual behavior

When no struct name is specified in `ts*` the program crashes.

### Steps to reproduce the behavior 

```
$ r2 /bin/ls
 -- Use +,-,*,/ to change the size of the block
[0x004049a0]> ts*
Segmentation fault (core dumped)
``````
# Additional Logs, screenshots, source-code,  configuration dump, ...

GDB backtrace, instruction, and registers

```
Program received signal SIGSEGV, Segmentation fault.
strlen () at ../sysdeps/x86_64/strlen.S:106
106	../sysdeps/x86_64/strlen.S: No such file or directory.
(gdb) bt
#0  strlen () at ../sysdeps/x86_64/strlen.S:106
#1  0x00007ffff3e0a049 in sdb_const_get_len (s=0x555555801c50,
    key=0x1 <error: Cannot access memory at address 0x1>, vlen=0x0, cas=0x0) at sdb.c:209
#2  0x00007ffff3e0a225 in sdb_const_get (s=0x555555801c50,
    key=0x1 <error: Cannot access memory at address 0x1>, cas=0x0) at sdb.c:254
#3  0x00007ffff7a8f5e2 in showFormat (core=0x55555575e580 <r>,
    name=0x1 <error: Cannot access memory at address 0x1>, mode=1) at cmd_type.c:164
#4  0x00007ffff7a914ef in cmd_type (data=0x55555575e580 <r>, input=0x555555ad6371 "s*")
    at cmd_type.c:661
#5  0x00007ffff7b14e20 in r_cmd_call (cmd=0x55555585b800, input=0x555555ad6370 "ts*")
    at cmd_api.c:237
#6  0x00007ffff7acae42 in r_core_cmd_subst_i (core=0x55555575e580 <r>, cmd=0x555555ad6370 "ts*",
    colon=0x0, tmpseek=0x7fffffffd936) at cmd.c:2902
#7  0x00007ffff7ac79b9 in r_core_cmd_subst (core=0x55555575e580 <r>, cmd=0x555555ad6370 "ts*")
    at cmd.c:1931
#8  0x00007ffff7acd5a9 in r_core_cmd (core=0x55555575e580 <r>, cstr=0x5555558e8a30 "ts*", log=1)
    at cmd.c:3606
#9  0x00007ffff7a35fb3 in r_core_prompt_exec (r=0x55555575e580 <r>) at core.c:2452
#10 0x000055555555b699 in main (argc=2, argv=0x7fffffffde28, envp=0x7fffffffde40)
    at radare2.c:1378
(gdb) x/5i $rip
=> 0x7ffff3810746 <strlen+38>:	movdqu xmm4,XMMWORD PTR [rax]
   0x7ffff381074a <strlen+42>:	pcmpeqb xmm4,xmm0
   0x7ffff381074e <strlen+46>:	pmovmskb edx,xmm4
   0x7ffff3810752 <strlen+50>:	test   edx,edx
   0x7ffff3810754 <strlen+52>:	je     0x7ffff381075a <strlen+58>
(gdb) i r
rax            0x1	1
rbx            0x0	0
rcx            0x1	1
rdx            0x0	0
rsi            0x1	1
rdi            0x1	1
rbp            0x7fffffffad40	0x7fffffffad40
rsp            0x7ffffffface8	0x7ffffffface8
r8             0xfec0c0c8	4274045128
r9             0x1999999999999999	1844674407370955161
r10            0x0	0
r11            0x4	4
r12            0x555555557300	93824992244480
r13            0x7fffffffde20	140737488346656
r14            0x0	0
r15            0x0	0
rip            0x7ffff3810746	0x7ffff3810746 <strlen+38>
eflags         0x10293	[ CF AF SF IF RF ]
cs             0x33	51
ss             0x2b	43
ds             0x0	0
es             0x0	0
fs             0x0	0
gs             0x0	0
```

### PR DESCRIPTION
**Pull request by fr0zn**

### CODE REVIEW

**Comment by radare** [comment]
This pr introduces many more issues. Please buld with asan and test properly. You cant access input+3 without checking that +1 and +2 are non nul

**Comment by XVilka** [comment]
Travis is red.

File: libr/core/cmd_type.c [hunk 1]
```diff
@@ -583,22 +583,29 @@ static int cmd_type(void *data, const char *input) {
...
+	case 'u': { // "tu"
+		char *tmp = strdup (input);
+		char *name = r_str_trim (strchr (tmp, ' '));
 		switch (input[1]) {
 		case '?':
 			r_core_cmd_help (core, help_msg_tu);
 			break;
 		case '*':
-			showFormat (core, input + 2, 1);
+			if (name != NULL) {
```
Comment (sivaramaaa) [inline libr/core/cmd_type.c]: `if (name) ` is enough , no need for `!= NULL` check , rest of it lgtm 

let's wait for tarvis !

File: libr/core/cmd_type.c [hunk 2]
```diff
@@ -583,22 +583,29 @@ static int cmd_type(void *data, const char *input) {
 		cmd_type_noreturn (core, input + 1);
 		break;
 	// t [typename] - show given type in C syntax
-	case 'u': // "tu"
+	case 'u': { // "tu"
+		char *tmp = strdup (input);
+		char *name = r_str_trim (strchr (tmp, ' '));
```
Comment (radare) [inline libr/core/cmd_type.c]: if tmp is null then it will be dereferenced by strchr and crash

Comment (radare) [reply libr/core/cmd_type.c]: name is always non NULL

File: libr/core/cmd_type.c [hunk 3]
```diff
@@ -583,22 +583,29 @@ static int cmd_type(void *data, const char *input) {
...
 		switch (input[1]) {
 		case '?':
 			r_core_cmd_help (core, help_msg_tu);
 			break;
 		case '*':
-			showFormat (core, input + 2, 1);
+			if (name) {
```
Comment (radare) [inline libr/core/cmd_type.c]: the name conditional is unnecessary

File: libr/core/cmd_type.c [hunk 4]
```diff
@@ -583,22 +583,29 @@ static int cmd_type(void *data, const char *input) {
...
+				showFormat (core, name, 1);
+			}
 			break;
 		case ' ':
-			showFormat (core, input + 2, 0);
+			if (name) {
```
Comment (radare) [inline libr/core/cmd_type.c]: same here

File: libr/core/cmd_type.c [hunk 5]
```diff
@@ -652,16 +659,21 @@ static int cmd_type(void *data, const char *input) {
 		}
 		break;
 	case 's': { // "ts"
-		char *name = strchr (input, ' ');
+		char *tmp = strdup (input);
+		char *name = r_str_trim (strchr (tmp, ' '));
```
Comment (radare) [inline libr/core/cmd_type.c]: this is not how r_str_trim works

Comment (radare) [reply libr/core/cmd_type.c]: you dont need to strdup anything. just skip the spaces if any, and this can be done in readonly.

Comment (radare) [reply libr/core/cmd_type.c]: char *name = r_str_trim_ro (input+1);

**Comment by radare** [comment]
Looks good now! Thanks

**Pull request merged**